src/libraries/System.Threading.RateLimiting/src/System/Threading/RateLimiting/TokenBucketRateLimiter.cs

```diff
@@ -126,7 +126,11 @@ protected override ValueTask<RateLimitLease> WaitAsyncCore(int tokenCount, Cance
                             RequestRegistration oldestRequest = _queue.DequeueHead();
                             _queueCount -= oldestRequest.Count;
                             Debug.Assert(_queueCount >= 0);
-                            oldestRequest.Tcs.TrySetResult(FailedLease);
+                            if (!oldestRequest.Tcs.TrySetResult(FailedLease))
+                            {
+                                // Updating queue count is handled by the cancellation code
```

Comment:
Was there an issue for this?

Reply:
Nope, found it when writing a test for PartitionedRateLimiter 😄